creme/activities/models/config.py

```diff
@@ -0,0 +1,152 @@
+################################################################################
+#    Creme is a free/open-source Customer Relationship Management software
+#    Copyright (C) 2023  Hybird
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+from __future__ import annotations
+
+import logging
+from datetime import time
+
+from django.db import models
+from django.utils.translation import gettext_lazy as _
+
+from creme.creme_core.models.auth import UserRole
+from creme.creme_core.models.base import CremeModel
+
+logger = logging.getLogger(__name__)
+
+
+class Weekday(models.IntegerChoices):
+    MONDAY = 1, _('Monday')
+    TUESDAY = 2, _('Tuesday')
+    WEDNESDAY = 3, _('Wednesday')
+    THURSDAY = 4, _('Thursday')
+    FRIDAY = 5, _('Friday')
+    SATURDAY = 6, _('Saturday')
+    SUNDAY = 7, _('Sunday')
+
+    @staticmethod
+    def default_days():
+        return (1, 2, 3, 4, 5, 6)
+
+
+class CalendarViewMode(models.TextChoices):
+    MONTH = 'month', _('Month')
+    WEEK = 'week', _('Week')
+
+
+class CalendarConfigItemManager(models.Manager):
+    def create_if_needed(self, *,
+                         role: UserRole | None,
+                         superuser: bool,
+                         **kwargs
+                         ) -> "CalendarConfigItem":
+        return self.get_or_create(
+            role=role,
+            superuser=superuser,
+            defaults=kwargs,
+        )[0]
```

Comment:
Useless, it's wrapper with not intelligence. Even in unit tests once you'll just rely on the default configuration create in populate.py I'm not sure it we'll be useful (and if it's useful, create a test case method).